temporal/fx.go

```diff
@@ -148,11 +148,14 @@ func NewServerFx(opts ...ServerOption) (*ServerFx, error) {
 		fx.Provide(ApplyClusterMetadataConfigProvider),
 		fx.Invoke(ServerLifetimeHooks),
 		FxLogAdapter,
+
+		fx.Populate(&s.blockingStart),
+		fx.Populate(&s.logger),
```

Comment:
The `ServerFx` object is constructed right here. `NewServerFxImpl` provides a `*ServerImpl`, not a `ServerFx` (the names in here are a twisty maze…)

`*ServerFx` implements `Server`, but `*ServerImpl` doesn't. It previously did, but the point was adding contexts, so it can't anymore.

I can rename stuff to make it less confusing but tbh I'm not even sure where to start :sweat_smile:  Maybe renaming `NewServerFxImpl` to `NewServerImpl`? Or `ServerImpl` to …? And then `ServerFx` to `ServerImpl`?

Reply:
...let's figure this out later lol